redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractBoxTest.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.layout
+
+import app.cash.redwood.layout.api.Constraint
+import app.cash.redwood.layout.widget.Box
+import app.cash.redwood.ui.Dp
+import app.cash.redwood.ui.dp
+import kotlin.test.Test
+
+abstract class AbstractBoxTest<T : Any> {
+
+  abstract fun Box(block: Box<T>.() -> Unit = {}): Box<T>
+
+  abstract fun Color(block: Color<T>.() -> Unit = {}): Color<T>
```

Comment:
Is there any advantage to doing this over calling `.apply` at the call sites below? I'm guessing each implementation is doing something like `.apply(block)`.

Reply:
Yep not much benefit - I'll update and move `apply` to the callsites.